### PR TITLE
Remove non promicro pins from converters

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Seems these have incorrectly slipped through.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h [hunk 1]
```diff
@@ -46,12 +46,6 @@
...
-#define D5 PAL_LINE(GPIOC, 14)
-#define C7 PAL_LINE(GPIOC, 15)
-#define F1 PAL_LINE(GPIOA, 0)
-#define F0 PAL_LINE(GPIOA, 1)
-
-// TX/RX pins of promicro
-#define B0 PAL_LINE(GPIOA, 9) // unconnected pin
+// LEDs
+#define D5 PAL_LINE(GPIOC, 14) // bottom row
+#define B0 PAL_LINE(GPIOA, 9)  // unconnected pin
```
Comment (daskygit) [inline platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h:51]: These are accessible LEDs on a promicro, so if the replacement implements them should there be a mapping?

Comment (zvecr) [reply platforms/chibios/converters/promicro_to_stemcell/_pin_defs.h:51]: We dont really have a decent way to stub out any gpio access so mapping to rando pins is the common practice. That way any LED use in the pre-converted code does not error out.

However in this particular case, it seems the stemcell has a single LED.

**Review by daskygit — APPROVED** [review]
I'm assuming we're happy to be merging to master?

**Review by megamind4089 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> I'm assuming we're happy to be merging to master?

I would say that it definitely qualifies as a bug fix, so yeah, it should.

**Comment by sadekbaroudi** [comment]
@drashna @megamind4089 @zvecr 

Why was this removed? Almost every keyboard I have using the stemcell uses these pins. What is the alternative to utilize these GPIO when using a stemcell?

**Comment by sadekbaroudi** [comment]
see https://github.com/qmk/qmk_firmware/pull/18236#issuecomment-1239787711 - I have the same issue with that as this one...

**Comment by zvecr** [comment]
> Almost every keyboard I have using the stemcell uses these pins

Then they are not "pro micro compatible" and your issue extends out from the currently implemented converter targets.

**Comment by sadekbaroudi** [comment]
> Then they are not "pro micro compatible" and your issue extends out from the currently implemented converter targets.

@zvecr - This doesn't address my point in the other PR ( 18236 )...

The architecture of "from pro micro" (which is implicit at this point, since the build syntax doesn't support a from parameter) doesn't make sense. You know the target hardware, you know the pinout. Your refusal to support the bottom pins is based on the underlying architecture, which is irrelevant for the consumer. They have X number of pins, they want to use them. Why limit it?

**Comment by zvecr** [comment]
But it is relevant to the consumer. How would a user use the converter feature to put a kb2040 into something that expects an elite c and use of its bottom row of pins? We know the pinout in both cases, but they are physically incompatible. The interface advertised, ensures that these scenarios fail for a predictable reason.

As mentioned in the other ticket, it does support a "from". The default is just currently "promicro" to maintain existing compatibility. Adding in an `elite_c -> stemcell` converter and updating `PIN_COMPATIBLE` for affected boards would be the way forward. The intention is that longer term, keyboards define what dev board they use. That way we can enforce someone trying to use converters on something like a dz60.

Outside of converters, there is no limit to pins.

**Comment by sadekbaroudi** [comment]
This translated into a discord conversation. For future reference (if needed):
https://discord.com/channels/440868230475677696/1017176320318373939/1017178935668265061

**Pull request merged**